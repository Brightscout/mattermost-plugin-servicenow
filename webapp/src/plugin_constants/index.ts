/**
* Keep all plugin related constants here
*/
export enum ToggleSwitchLabelPositioning {
    Left = 'left',
    Right = 'right',
}

export const DOWNLOAD_UPDATE_SET_LINK = '/download';

const MMCSRF = 'MMCSRF';
const HeaderCSRFToken = 'X-CSRF-Token';
const MMUSERID = 'MMUSERID';
const RightSidebarHeader = 'Subscriptions';
const RhsToggleLabel = 'Show all subscriptions';
const InvalidAutoCompleteValueMsg = 'Invalid value, please select a value from the suggestions.';
const ChannelHeaderTooltipText = 'ServiceNow';
const DefaultCharThresholdToShowSuggestions = 4;
const DefaultPage = 0;
const DefaultPageSize = 100;
const PrivateChannelType = 'P';

export enum SubscriptionEvents {
    state = 'state',
    priority = 'priority',
    commented = 'commented',
    assignedTo = 'assigned_to',
    assignmentGroup = 'assignment_group',
}

// Used in search records panel for rendering the key-value pairs of the record for showing the record details
const RecordDataLabelConfig: RecordDataLabelConfigType[] = [
    {
        key: 'short_description',
        label: 'Short Description',
    }, {
        key: 'state',
        label: 'State',
    }, {
        key: 'priority',
        label: 'Priority',
    }, {
        key: 'assigned_to',
        label: 'Assigned To',
    }, {
        key: 'assignment_group',
        label: 'Assignment Group',
    },
];

// Plugin api service (RTK query) configs
const pluginApiServiceConfigs: Record<ApiServiceName, PluginApiService> = {
    getChannels: {
        path: '/channels',
        method: 'GET',
        apiServiceName: 'getChannels',
    },
    searchRecords: {
        path: '/records',
        method: 'GET',
        apiServiceName: 'searchRecords',
    },
    getRecord: {
        path: '/records',
        method: 'GET',
        apiServiceName: 'getRecord',
    },
    createSubscription: {
        path: '/subscriptions',
        method: 'POST',
        apiServiceName: 'createSubscription',
    },
    fetchSubscriptions: {
        path: '/subscriptions',
        method: 'GET',
        apiServiceName: 'fetchSubscriptions',
    },
};

export const PanelDefaultHeights = {
    channelPanel: 151,
    recordTypePanel: 195,
    searchRecordPanel: 203,
    searchRecordPanelExpanded: 372,
    eventsPanel: 500,
    successPanel: 220,
    panelHeader: 65,
};

export default {
    RightSidebarHeader,
    DOWNLOAD_UPDATE_SET_LINK,
    pluginApiServiceConfigs,
    MMCSRF,
    HeaderCSRFToken,
    InvalidAutoCompleteValueMsg,
    RecordDataLabelConfig,
    MMUSERID,
    ChannelHeaderTooltipText,
    RhsToggleLabel,
    DefaultCharThresholdToShowSuggestions,
    DefaultPage,
    DefaultPageSize,
    PrivateChannelType,
};
