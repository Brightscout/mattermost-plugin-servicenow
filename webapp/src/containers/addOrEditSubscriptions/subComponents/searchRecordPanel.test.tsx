import React from 'react';
import {Provider} from 'react-redux';

import {shallow, ShallowWrapper} from 'enzyme';

import plugin_constants from 'plugin_constants';
import {configureStore} from '../../../../tests/setup';

import SearchRecordsPanel from './searchRecordsPanel';

const mockOnContinue = jest.fn();
const mockOnBack = jest.fn();
const mockSetRecordValue = jest.fn();
const mockSetSuggestionChosen = jest.fn();
const mockSetApiError = jest.fn();
const mockSetApiResponseValid = jest.fn();
const mockSetShowModalLoader = jest.fn();
const mockSetRecordId = jest.fn();
const mockSetResetStates = jest.fn();

const mockRecordType: RecordType = 'incident';

const searchRecordPanelProps = {
    className: 'mockClassName',
    error: 'mockError',
    onBack: mockOnBack,
    onContinue: mockOnContinue,
    actionBtnDisabled: true,
    requiredFieldValidationErr: true,
    recordValue: 'mockRecordValue',
    setRecordValue: mockSetRecordValue,
    suggestionChosen: true,
    setSuggestionChosen: mockSetSuggestionChosen,
    recordType: mockRecordType,
    setApiError: mockSetApiError,
    setApiResponseValid: mockSetApiResponseValid,
    setShowModalLoader: mockSetShowModalLoader,
    recordId: 'mockRecordId',
    setRecordId: mockSetRecordId,
    resetStates: true,
    setResetStates: mockSetResetStates,
};

describe('Search Record Panel', () => {
    let component: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
    const initialState = {};
    const mockStore = configureStore();

    beforeEach(() => {
        const store = mockStore(initialState);
        component = shallow(
            <Provider store={store}>
                <SearchRecordsPanel {...searchRecordPanelProps}/>
            </Provider>,
        );
    });

    afterEach(() => {
        component.unmount();
    });

    it('Should render correctly', () => {
        expect(component).toMatchSnapshot();
    });
});
