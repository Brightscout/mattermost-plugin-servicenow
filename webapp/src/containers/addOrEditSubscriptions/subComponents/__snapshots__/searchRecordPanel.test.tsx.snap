// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Record Panel Should render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <ForwardRef
    actionBtnDisabled={true}
    className="mockClassName"
    error="mockError"
    onBack={[MockFunction]}
    onContinue={[MockFunction]}
    recordId="mockRecordId"
    recordType="incident"
    recordValue="mockRecordValue"
    requiredFieldValidationErr={true}
    resetStates={true}
    setApiError={[MockFunction]}
    setApiResponseValid={[MockFunction]}
    setRecordId={[MockFunction]}
    setRecordValue={[MockFunction]}
    setResetStates={[MockFunction]}
    setShowModalLoader={[MockFunction]}
    setSuggestionChosen={[MockFunction]}
    suggestionChosen={true}
  />
</ContextProvider>
`;
